### ISSUE TITLE
Possibility to remove the icons

### ISSUE THREAD

**Issue by Nathan-Furnal** [emacs-lsp/lsp-treemacs#66]
Hi,

I'm using lsp-treemacs and it added icons which I don't use at all in regular treemacs. 
How can I remove the icons and not use them?

Thanks!

**Comment by nbfalcon** [comment]
How did you disable the icons for regular treemacs? You can customize the icon theme of lsp-treemacs using the `lsp-treemacs-theme` variable, so perhaps you could set some icon theme that has no icons for it.

**Comment by Nathan-Furnal** [comment]
In treemacs one can set a variable like this : 

```elisp
(setq treemacs-no-png-images t)
```
I'm not sure how changing the theme would affect the behavior of my current theme, it'd be just easier not to have them tbh

**Comment by nbfalcon** [comment]
If you are happy with sacrificing the arrows, I could offer the following (temporary!) kludge, until we actually implement this properly (so don't close this issue yet):

```emacs-lisp
(eval-after-load 'lsp-treemacs (defun lsp-treemacs--generic-icon (&rest _) ""))
```

Here is how it would look on latest treemacs master + lsp-mode master:
![20201112-180325](https://user-images.githubusercontent.com/69264154/98971444-73a81880-2511-11eb-8567-7092c65b4f6c.png)

**Comment by swinkels** [comment]
> In treemacs one can set a variable like this :
> 
> ```emacs-lisp
> (setq treemacs-no-png-images t)
> ```
> 
> I'm not sure how changing the theme would affect the behavior of my current theme, it'd be just easier not to have them tbh

I see that `lsp-treemacs.el` always calls `treemacs-get-icon-value` with second parameter `nil`. That second parameter indicates whether the TUI version of the icon should be used (when non-nil) or the GUI version (when `nil`). I may be wrong here as I'm not familiar with the treemacs code, or Emacs Lisp for that matter, but `lsp-treemacs` could take the value of `treemacs-no-png-images` into account. That it doesn't appear to so may be by design, I don't know.

To disable the use of icons I locally replaced the `nil` in the call to `treemacs-get-icon-value` by `treemacs-no-ping-images`, which I've set to `t`. It works for the use case I have.

**Comment by eeshugerman** [comment]
@nbfalcon if implemented, would this proposed feature/switch also solve https://github.com/hlissner/emacs-doom-themes/issues/500? or shall i open a separate issue?

**Comment by nbfalcon** [comment]
No, that is unrelated (this is `lsp-treemacs`, not normal `treemacs`)

**Comment by eeshugerman** [comment]
It was determined that the issue only appears once `lsp-mode` starts up, so I'm pretty sure it's an `lsp-treemacs` issue.

**Comment by nbfalcon** [comment]
Upon further inspection, this is probably related. Can you check with the `triage/ugly-icons` branch on my fork?

**Comment by eeshugerman** [comment]
it works! no icon issues present when i use that branch.

**Comment by nbfalcon** [comment]
Great! So the issue here is that we are modifying the "Default" treemacs theme, and replacing some icons which were probably provided by `all-the-icons` with `png` icons. The former uses a font and so they are vector graphics and scale well, while the latter ones are raster graphics and don't. The icons that are needlessly overridden in `lsp-treemacs-icons` would need to be examined. Note that this is unrelated to *this* issue, though.

**Comment by eeshugerman** [comment]
got it :+1: i opened a new issue, see above
